### DISCUSSION TITLE
[OKD FCOS 4.15] Change Master and Worker IP Address to another subnet range

### DISCUSSION THREAD

**Discussion by parseltongued** [okd-project/okd#1994] [General]
**Describe the bug**
I have an existing 3-node OKD FCOS 4.15 airgap cluster and i'm looking to export the entire cluster to a different IP subnet range (Old IP range: 172.17.12.* to New IP range: 192.168.20.*) but no fqdn change.  My on-prem dns and dhcp servers, "Mirror registry for Red Hat OpenShift" and HAProxy are working on the new IP address range. The 3 master nodes have new IP addresses however they are not starting up (HAProxy 6443, 22623 is a flat red).

Although i've installed the cluster by specifying fqdn, suspecting it used the resolved IP to create the SSL cert, if so I need to recreate them, but I can't use oc commands because 6443 is not up.

How should I proceed and is there a proven procedure to change entire OKD cluster subnet?

**Cluster environment**
OKD Cluster Version: 4.15.0-0.okd-2024-03-10-010116
Kernel version: v1.28.2-3598+6e2789bbd58938-dirty
Installation method: Bare-metal UPI (Airgapped, self hosted quay)

**Comment by melledouwsma** [comment]
To my knowledge, there is no proven procedure for changing the IP addresses of all nodes in a cluster. Would it be possible to deploy three new nodes within the new IP range and then remove the existing nodes from the cluster? The contents of [How to replace unhealthy master nodes in UPI on OpenShift 4.x](https://access.redhat.com/solutions/6387841) may also be relevant in this case.

**Reply by parseltongued** [reply]
thank you. After reverting my VMs I am now facing another bigger issue would raise as another bug.
